File: Luogu/P1007.py
l = int(input())
n = int(input())

if n == 0:
    print(0, 0)
else:
    pos = map(int, input().split())

    min_t = -1
    max_t = -1

    for i in pos:
        if min(i, l + 1 - i) > min_t:
            min_t = min(i, l + 1 - i)
        if max(i, l + 1 - i) > max_t:
            max_t = max(i, l + 1 - i)

    print(min_t, max_t)

File: Luogu/P1007.cpp
#include <stdio.h>

int min(int a, int b) {
	return a < b ? a : b;
}
int max(int a, int b) {
	return a > b ? a : b;
}

const int MAXN = 5010;
int arr[MAXN];

int main() {
	int l, n;
	scanf("%d%d", &l, &n);
	for (int i = 0; i < n; i++) {
		scanf("%d", &arr[i]);
	}
	
	int min_t = 0;
	int max_t = 0;
	
	for (int i = 0; i < n; i++) {
		if (min(arr[i], l + 1 - arr[i]) > min_t) {
			min_t = min(arr[i], l + 1 - arr[i]);
		}
		if (max(arr[i], l + 1 - arr[i]) > max_t) {
			max_t = max(arr[i], l + 1 - arr[i]);
		}
	}
	
	printf("%d %d", min_t, max_t);
}
